src/TraceEvent/Computers/StartStopThreadTimeComputer.cs

```diff
@@ -0,0 +1,229 @@
+        {}
+
+        protected override void Subscribe(TraceLogEventSource eventSource)
+        {
+            eventSource.Clr.WaitHandleWaitStart += OnStart;
+            eventSource.Clr.WaitHandleWaitStop += OnStop;
+        }
+
+        protected override void RecordAdditionalDataOnMissingStartAtTraceStart(StackSourceSample sample)
+        {
+            sample.StackIndex = _stackSource.Interner.CallStackIntern(_stackSource.Interner.FrameIntern($"EventData DurationNs {sample.Metric*NanosInMillisecond:N0}"), sample.StackIndex);
+        }
+
+        protected override void RecordAdditionalStartData(StackSourceSample sample, TraceEvent data)
+        {
+            var startData = (WaitHandleWaitStartTraceData) data;
+            sample.StackIndex = _interner.CallStackIntern(_interner.FrameIntern($"EventData WaitSource {startData.WaitSource}"), sample.StackIndex);
+            sample.StackIndex = _interner.CallStackIntern(_interner.FrameIntern($"EventData AssociatedObjectID {startData.AssociatedObjectID}"), sample.StackIndex);
+        }
+
+        protected override void RecordAdditionalDataOnMissingStopAtTraceEnd(StackSourceSample sample)
+        {
+            sample.StackIndex = _stackSource.Interner.CallStackIntern(_stackSource.Interner.FrameIntern($"EventData DurationNs {sample.Metric*NanosInMillisecond:N0}"), sample.StackIndex);
+        }
+
+        protected override void RecordAdditionalStopData(StackSourceSample sample, TraceEvent data)
+        {
+            var stopData = (WaitHandleWaitStopTraceData) data;
+            var durationMs = stopData.TimeStampRelativeMSec - sample.TimeRelativeMSec; // recompute the duration to get better precision than float
+            sample.StackIndex = _stackSource.Interner.CallStackIntern(_stackSource.Interner.FrameIntern($"EventData DurationNs {durationMs*NanosInMillisecond:N0}"), sample.StackIndex);
+        }
+    }
+    
+    public abstract class StartStopThreadTimeComputer
+    {
+        protected readonly TraceLog _eventLog;
+        protected readonly MutableTraceEventStackSource _stackSource;
+        protected readonly StackSourceInterner _interner;
+        
+        private readonly StackSourceSample[] _samplesPerThread;
+
+        protected const double NanosInMillisecond = 1000 * 1000;
+        private const string BrokenEventNoStart = "BROKEN_EVENT NO_CORRESPONDING_START";
+        private const string BrokenEventNoStop = "BROKEN_EVENT NO_CORRESPONDING_STOP";
+
+        protected StartStopThreadTimeComputer(TraceLog eventLog, MutableTraceEventStackSource stackSource)
+        {
+            _eventLog = eventLog;
+            _stackSource = stackSource;
+            _interner = stackSource.Interner;
+            _samplesPerThread = new StackSourceSample[eventLog.Threads.Count];
+        }
+
+        protected abstract void Subscribe(TraceLogEventSource eventSource);
```

Comment:
Can you please add comments for each of these methods to make clear what can/should happen in their implementations?